source/components/Timeago/index.js

```diff
@@ -1,70 +1,81 @@
-        return date.toLocaleDateString();
-    }
-
-    if (diffInSeconds > 86400) {
-        // more than a day ago
-        return `${Math.round(diffInSeconds / 60 / 60 / 24)}d`;
-    }
-
-    if (diffInSeconds > 3600) {
-        // more than an hour ago
-        return `${Math.round(diffInSeconds / 60 / 60)}h`;
-    }
-
-    if (diffInSeconds < 60) {
-        // less than a minute ago
-        return 'now';
-    }
-
-    return `${Math.round(diffInSeconds / 60)}m`;
-}
+export const TIMEAGO_SHOW_DATE_THRESHOLD = FIVE_DAYS_SECONDS;
+export const TIMEAGO_NOW_THRESHOLD = MINUTE_SECONDS;
 
 /**
  * The Timeago component is a small component that
  * shows the number of seconds/minutes/days from given datetime.
- *
- * It all happens after the component is mounted so it's safe to use this
- * component on the Back-End without messing up the hydration.
  */
-export default class Timeago extends React.Component {
-    state = {
-        display: this.props.datetime,
-    };
+export default function Timeago({
+    datetime, renderNow, renderDate, renderDays, renderHours, renderMinutes,
+}) {
+    const date = new Date(datetime);
 
-    static propTypes = {
-        datetime: PropTypes.oneOfType([
-            PropTypes.instanceOf(Date),
-            PropTypes.string,
-        ]).isRequired,
-    };
+    const withCustomRender = (renderFunction, value, defaultElement = value) => (typeof (renderFunction) === 'function' ? renderFunction(value) : defaultElement);
+
+    const renderTime = () => {
+        const now = Date.now();
+        const diffInSeconds = (now - date.getTime()) / SECOND_MILLISECONDS;
 
-    componentDidMount() {
-        const { datetime } = this.props;
+        if (diffInSeconds > TIMEAGO_SHOW_DATE_THRESHOLD) {
+            const dateString = date.toLocaleDateString();
 
-        this.setState({
-            display: getTimeDistanceString(datetime),
-        });
-    }
+            return withCustomRender(renderDate, dateString);
+        }
 
-    render() {
-        const { datetime } = this.props;
-        const { display } = this.state;
+        if (diffInSeconds > DAY_SECONDS) {
+            // more than a day ago
+            const days = Math.round(diffInSeconds / DAY_SECONDS);
+            return withCustomRender(renderDays, days, `${days}d`);
+        }
 
-        return (
-            <span title={datetime}>
-                {display}
-            </span>
-        );
-    }
+        if (diffInSeconds > HOUR_SECONDS) {
+            // more than an hour ago
+            const hours = Math.round(diffInSeconds / HOUR_SECONDS);
+            return withCustomRender(renderHours, hours, `${hours}h`);
+        }
+
+        if (diffInSeconds > TIMEAGO_NOW_THRESHOLD) {
+            const minutes = Math.round(diffInSeconds / MINUTE_SECONDS);
+            return withCustomRender(renderMinutes, minutes, `${minutes}m`);
+        }
+
+        // less than a minute ago
+        return typeof (renderNow) === 'function' ? renderNow() : 'now';
```

Comment:
Shouldn't this be translated?

Reply:
https://github.com/Wikia/react-common/pull/182#discussion_r473783212